mailpoet/lib/Segments/WooCommerce.php

```diff
@@ -338,18 +338,43 @@ private function updateNames(array $orders): void {
     }
     $subscribersTable = $this->entityManager->getClassMetadata(SubscriberEntity::class)->getTableName();
 
-    $metaKeys = [
-      '_billing_first_name',
-      '_billing_last_name',
-    ];
-    $metaData = $this->connection->executeQuery("
-      SELECT post_id, meta_key, meta_value
-      FROM {$wpdb->postmeta}
-      WHERE meta_key IN (:metaKeys) AND post_id IN (:postIds)
-    ",
-      ['metaKeys' => $metaKeys, 'postIds' => array_values($orders)],
-      ['metaKeys' => Connection::PARAM_STR_ARRAY, 'postIds' => Connection::PARAM_INT_ARRAY]
-    )->fetchAllAssociative();
+    if ($this->woocommerceHelper->isWooCommerceCustomOrdersTableEnabled()) {
+      $addressesTableName = $this->woocommerceHelper->getAddressesTableName();
```

Comment:
Given that the method is already rather big and we are dealing with a paradigm shift on WC side, I was wondering what you think about creating a service dedicated to HPOS and using its methods instead of adding the changes directly where needed and making the logic even bigger?

Reply:
Thanks for the review, @samnajian. I'm not against the refactor that you are suggesting but at the same time I don't see a significant advantage in it, and considering that this is the last piece that is missing to support HPOS in this class, I'm inclined to prefer to not make this change at this point as it would mean revisiting all the other code changes that were already made in this class. This especially considering that ideally I would like to finish this project this week. Let me know your thoughts.